### PR TITLE
RavenDB-12094

### PR DESCRIPTION
**Pull request by aviv86**

### CODE REVIEW

**Review by ayende — APPROVED** [review]

File: src/Raven.Server/Documents/Patch/ScriptRunner.cs [hunk 1]
```diff
@@ -173,7 +175,7 @@ public SingleRun(DocumentDatabase database, RavenConfiguration configuration, Sc
                 ScriptEngine.SetValue("Raven_Min", new ClrFunctionInstance(ScriptEngine, Raven_Min));
                 ScriptEngine.SetValue("Raven_Max", new ClrFunctionInstance(ScriptEngine, Raven_Max));
 
-                ScriptEngine.SetValue("convertJsTimeToTimeSpanString", new ClrFunctionInstance(ScriptEngine, ConvertJsTimeToTimeSpanString));
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Patch/ScriptRunner.cs]: This means that old client libraries will stop working due to the lack of this method?

File: src/Raven.Server/Documents/Patch/ScriptRunner.cs [hunk 2]
```diff
@@ -926,6 +976,17 @@ private static JsValue ToStringWithFormat(JsValue self, JsValue[] args)
                         num.ToString(cultureInfo);
                 }
 
+                if (args[0].IsString())
+                {
+                    if (DateTime.TryParseExact(args[0].AsString(), "o", CultureInfo.InvariantCulture, DateTimeStyles.RoundtripKind, out var date) == false)
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Patch/ScriptRunner.cs]: I think that instead of 'o' we should use 'DefaultFormat.DateTimeFormatsToRead' ?

Comment (ppekrol) [inline src/Raven.Server/Documents/Patch/ScriptRunner.cs]: Maybe we should use LazyStringParser instead?

**Review by ppekrol — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Documents/Patch/ScriptRunner.cs (hunk 1 above)
Comment (ayende) [reply src/Raven.Server/Documents/Patch/ScriptRunner.cs]: I think so, I believe we need to keep it around.

File: src/Raven.Server/Documents/Patch/ScriptRunner.cs (hunk 2 above)
Comment (ppekrol) [reply src/Raven.Server/Documents/Patch/ScriptRunner.cs]: @aviv86 can we use LazyStringParser here?

**Review by ppekrol — APPROVED** [review]

**Pull request merged**